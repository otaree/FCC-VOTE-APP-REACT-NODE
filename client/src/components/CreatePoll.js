import React from 'react';
import { connect } from 'react-redux';

import * as pollActions from '../store/actions/poll';

export class CreatePoll extends React.Component {
    state = {
        question: "",
        options: [{
            value: ""
        }, {
            value: ""
        }]
    };

    questionChangeHandler = e => {
        this.setState({ question: e.target.value });
    };

    optionsChangeHandler = (e, index) => {
        const value = e.target.value;
        this.setState(prevState => {
            return {
                options: prevState.options.map((option, i) => {
                    if (index === i) {
                        return {
                            value
                        }
                    } else {
                        return option;
                    }
                })
            };
        });
    };

    render() {
        let createPoll = (
            <div>
                <form>
                    <div>
                        <input 
                            type="text"
                            placeholder="Question"
                            value={this.state.question}
                            onChange={this.questionChangeHandler}
                        />
                    </div>
                    <div>
                        {
                            this.state.options.map((option, index) => {
                                return (
                                    <input
                                        key={index}
                                        type="text" 
                                        placeholder={`Option ${index + 1}`}
                                        value={option.value}
                                        onChange={e => this.optionsChangeHandler(e, index)}
                                    />
                                )
                            })
                        }
                        <button>Add Option</button>
                    </div>
                    <div>
                        <button>Create Poll</button>
                    </div>
                </form>
            </div>
        );
        return createPoll;
    }
}

const mapStateToProps = state => {
    return {
        loading: state.poll.loading,
        token: state.auth.token
    };
};

const mapDispatchToProps = dispatch => {
    return {
        createPoll: (poll, token) => pollActions.createPoll(poll, token)
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CreatePoll);