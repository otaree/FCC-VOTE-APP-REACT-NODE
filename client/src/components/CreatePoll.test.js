import React from 'react';
import { shallow } from 'enzyme';

import { CreatePoll } from './CreatePoll';

describe('CreatePoll', () => {
    const createPoll = shallow(<CreatePoll />);

    it('renders correctly', () => {
        expect(createPoll).toMatchSnapshot();
    });
});