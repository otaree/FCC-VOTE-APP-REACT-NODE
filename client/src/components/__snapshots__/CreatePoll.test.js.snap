// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoll renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreatePoll />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div>
                <input
                        onChange={[Function]}
                        placeholder="Question"
                        type="text"
                        value=""
                />
        </div>
        <div>
                <input
                        onChange={[Function]}
                        placeholder="Option 1"
                        type="text"
                        value=""
                />
                <input
                        onChange={[Function]}
                        placeholder="Option 2"
                        type="text"
                        value=""
                />
                <button>
                        Add Option
                </button>
        </div>
        <div>
                <button>
                        Create Poll
                </button>
        </div>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <input
                        onChange={[Function]}
                        placeholder="Question"
                        type="text"
                        value=""
            />
</div>,
          <div>
            <input
                        onChange={[Function]}
                        placeholder="Option 1"
                        type="text"
                        value=""
            />
            <input
                        onChange={[Function]}
                        placeholder="Option 2"
                        type="text"
                        value=""
            />
            <button>
                        Add Option
            </button>
</div>,
          <div>
            <button>
                        Create Poll
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Question"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Question",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <input
                  onChange={[Function]}
                  placeholder="Option 1"
                  type="text"
                  value=""
/>,
                <input
                  onChange={[Function]}
                  placeholder="Option 2"
                  type="text"
                  value=""
/>,
              ],
              <button>
                Add Option
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Option 1",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Option 2",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Option",
              },
              "ref": null,
              "rendered": "Add Option",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              Create Poll
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Poll",
            },
            "ref": null,
            "rendered": "Create Poll",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div>
                    <input
                              onChange={[Function]}
                              placeholder="Question"
                              type="text"
                              value=""
                    />
          </div>
          <div>
                    <input
                              onChange={[Function]}
                              placeholder="Option 1"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="Option 2"
                              type="text"
                              value=""
                    />
                    <button>
                              Add Option
                    </button>
          </div>
          <div>
                    <button>
                              Create Poll
                    </button>
          </div>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <input
                            onChange={[Function]}
                            placeholder="Question"
                            type="text"
                            value=""
              />
</div>,
            <div>
              <input
                            onChange={[Function]}
                            placeholder="Option 1"
                            type="text"
                            value=""
              />
              <input
                            onChange={[Function]}
                            placeholder="Option 2"
                            type="text"
                            value=""
              />
              <button>
                            Add Option
              </button>
</div>,
            <div>
              <button>
                            Create Poll
              </button>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                onChange={[Function]}
                placeholder="Question"
                type="text"
                value=""
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onChange": [Function],
                "placeholder": "Question",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  <input
                    onChange={[Function]}
                    placeholder="Option 1"
                    type="text"
                    value=""
/>,
                  <input
                    onChange={[Function]}
                    placeholder="Option 2"
                    type="text"
                    value=""
/>,
                ],
                <button>
                  Add Option
</button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Option 1",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "onChange": [Function],
                  "placeholder": "Option 2",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Option",
                },
                "ref": null,
                "rendered": "Add Option",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button>
                Create Poll
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create Poll",
              },
              "ref": null,
              "rendered": "Create Poll",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
